drivers/periph_common/flashpage.c

```diff
@@ -104,4 +106,79 @@ int flashpage_rwwee_write_and_verify(unsigned page, const void *data)
 
 #endif
 
+/* We have to write whole words, but writing 0xFF is basically a no-op
+ * so fill the unaligned bytes with 0xFF to get a whole extra word.
+ * To support writes of data with less than 4 bytes, an offset needs
+ * to be supplied.
+ */
+static void unaligned_pad_start(const void *dst, uint8_t *buffer,
+                                const void *data, uint8_t len, uint8_t offset)
+{
+    if (IS_ACTIVE(FLASHPAGE_NO_BLIND_WRITES)) {
+        memcpy(buffer, dst, FLASHPAGE_WRITE_BLOCK_SIZE);
```

Comment:
This assumes that flash is mapped into the data memory, which is not universally true.

I'm still convinced that we should have some `rom_memcpy()`, `rom_strcpy()` etc. that are `static inline` wrappers to `memcpy()`, `strcpy()`, etc. for ARM, but allow using PROGMEM on AVR and Xtensa. Those would make this work universally. (But in absence of them, we cannot do anything for more classic Harvard machines that do not allow blind writes.)